lbs/storage.c

```diff
@@ -283,6 +283,9 @@ storage_write(struct storage_state * S,
 	uint64_t fnum;
 	char * s = NULL;	/* free(NULL) simplifies error path. */
 
+	/* Sanity check that nblks * S->blocklen <= SIZE_MAX. */
+	assert(nblks != 0 && S->blocklen <= SIZE_MAX / nblks);
```

Comment:
I would write this as `nblks <= SIZE_MAX / S->blocklen` since that way we're checking both sides of `nblks`.  (Also theoretically the division could be precomputed that way, not that I'm suggesting we do that right now.)